sonic-net/sonic-mgmt
# warm-reboot CPA traffic handling test gap

### Description of PR
Fix test gap for CPA feature.
https://github.com/sonic-net/sonic-mgmt/issues/4284

Add following test case to cover.

```
Test case 1 – non broadcast ARP Reply
Test objective: Verify ARP Reply for known mac is not broadcast to server during warm-reboot
Test steps:
   1. Trigger Warm-Reboot, PTF send ARP Request from Port Ethernet-A
   2. Verify ARP Reply received only on Ethernet-A (non-broadcast)
   3. Check if Warm-Reboot timeout, and finish test case post check

Test case 2 – ARP Request to ferret server
Test objective: Verify during Warm-Reboot, ARP Request from server send to ferret
Test steps:
  1. Trigger Warm-Reboot
  2. Start new thread and in this thread let PTF periodically send ARP Request from Port Ethernet-A, check send counter X.
  3. Wait and show everflow ACL counter
  4. Verify X matches everflow acl counter increase, verify ARP Reply received from Ethernet-A
  5. Check if Warm-Reboot timeout, and finish test case post check (join new thread ...)

Test Case 3 – ARP Reply from ferret server During Warm-Reboot
Test objective: Verify during Warm-Reboot, ARP Reply all comes from ferret server and in vxlan encap
Test steps:
  1. Clear related interface counters, vxlan counter, Everflow ACL counters
  2. Trigger Warm-Reboot, PTF send ARP Request from Port Ethernet-A, check send counter X
  3. Show everflow ACL counter match send counter X, PTF received ARP Reply from Ethernet-A and counter matches X (refer to Note)
  4. Verify vxlan packets counter matches X (refer to Note)
  5. Check if Warm-Reboot timeout, and finish test case post check

Note: 
It's hard to determine when warm-reboot starts, so counter send counter X may always large than everflow ACL and VXLAN counters, to simplify the test case, we can just monitor the increase value of acl and vxlan counters)
Due to show vxlan counter not work, currently only check vxlan tunnel status during warm-reboot
```

Summary:
Fixes #4284 

### Type of change



### Back port request

### Approach
#### What is the motivation for this PR?
Fix test gap

#### How did you do it?
Add test case for address resolution traffic during warm-reboot

#### How did you verify/test it?
Verify on DUT
